src/main/kotlin/no/nav/klage/oppgave/service/TilgangService.kt

```diff
@@ -0,0 +1,86 @@
+package no.nav.klage.oppgave.service
+
+import no.nav.klage.oppgave.clients.egenansatt.EgenAnsattService
+import no.nav.klage.oppgave.clients.pdl.PdlFacade
+import no.nav.klage.oppgave.domain.AuditLogEvent
+import no.nav.klage.oppgave.domain.AuditLogEvent.Decision.ALLOW
+import no.nav.klage.oppgave.domain.AuditLogEvent.Decision.DENY
+import no.nav.klage.oppgave.exceptions.MissingTilgangException
+import no.nav.klage.oppgave.repositories.InnloggetSaksbehandlerRepository
+import no.nav.klage.oppgave.util.AuditLogger
+import no.nav.klage.oppgave.util.getLogger
+import no.nav.klage.oppgave.util.getSecureLogger
+import org.springframework.stereotype.Service
+
+@Service
+class TilgangService(
+    private val pdlFacade: PdlFacade,
+    private val egenAnsattService: EgenAnsattService,
+    private val innloggetSaksbehandlerRepository: InnloggetSaksbehandlerRepository,
+    private val auditLogger: AuditLogger
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+        private val securelogger = getSecureLogger()
+    }
+
+    private fun auditLog(fnr: String, decision: AuditLogEvent.Decision) {
+        auditLogger.log(
+            AuditLogEvent(
+                navIdent = innloggetSaksbehandlerRepository.getInnloggetIdent(),
+                action = AuditLogEvent.Action.KLAGEBEHANDLING_VIEW,
+                decision = decision,
+                personFnr = fnr
+            )
+        )
+    }
+
+    fun verifySaksbehandlersTilgangTil(fnr: String) {
+        if (!harSaksbehandlerTilgangTil(fnr)) {
+            throw MissingTilgangException("Not authorized to access this user")
+        }
+    }
+
+    fun harSaksbehandlerTilgangTil(fnr: String): Boolean {
+        val personInfo = pdlFacade.getPersonInfo(fnr)
+        if (personInfo == null) {
+            //TODO: Hva defaulter vi til? Velger å ikke deny nå, så får vi se..
+            securelogger.error("PDL didnt return anything for fnr $fnr, we default to give access")
```

Comment:
I dev kanskje dette kan være nødvendig iaf for å få ut noe.

Reply:
Ja, det var det jeg tenkte også.